test/ServiceTest.cpp

```diff
@@ -605,3 +609,33 @@ TEST_F(ServiceTest, bindingToTripleComponent) {
           {{"type", "INVALID_TYPE"}, {"value", "v"}}),
       ::testing::HasSubstr("Type INVALID_TYPE is undefined"));
 }
+
+TEST_F(ServiceTest, idToValueForValuesClause) {
+  auto idToVc = Service::idToValueForValuesClause;
+  LocalVocab localVocab{};
+  auto index = ad_utility::testing::makeIndexWithTestSettings();
+
+  // undefined, blanknode -> nullopt
+  EXPECT_EQ(idToVc(index, Id::makeFromBlankNodeIndex(BlankNodeIndex::make(0)),
+                   localVocab),
+            std::nullopt);
+  EXPECT_EQ(idToVc(index, Id::makeUndefined(), localVocab), std::nullopt);
```

Comment:
That is also wrong, for undef you have to write the string `"UNDEF"` in the `VALUES` clause.